04_numeronym/brvishnu92/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	x := []string{}
+
+	for _, val := range vals {
+		end := len(val)
+		if end > 3 {
+			x = append(x, fmt.Sprintf("%v%v%v", val[:1], end-2, val[end-1:]))
```

Comment:
Personally, I prefer specifying actual formatting like `%c%d%c` as type is known. Then if an incorrect argument is provided, it will give a compile-time error.

Reply:
@kasunfdo  I did try %c but it doesnt work as my value returns a slice.

Sprintf format %c has arg val[:1] of wrong type string

Reply:
Sorry, my bad. I forgot to mention that you should use `val(0), end-2, val[end-1]`

Reply:
@kasunfdo Oh yeah, sorry thanks for that :-)

Reply:
@kasunfdo Can you please approve the changes? I am unable to merge.